### PR TITLE
Add net.jammr.jammr

### PR DESCRIPTION
**Pull request by stefanha**
jammr allows musicians to jam together online (https://jammr.net/).
This app contains the GPL-2.0-or-later licensed client which is a Qt
application developed by the upstream Wahjam project
(https://wahjam.org/).  I am the owner of jammr and the maintainer of
Wahjam.

The online jamming platform that the client connects to is free to use.
A Premium subscription is available offering additional features.  I
think this app fits within the scope of Flathub since it is open source
and can be used for free.

Add a manifest for jammr 1.2.8 using org.kde.Platform 5.14.  Luckily
most dependencies are already present in org.kde.Sdk.  Add PortAudio and
PortMidi modules similar to how the org.mixxx.Mixxx manifest does it.

The client needs access to ALSA, PulseAudio, and/or JACK audio devices.
It also uses networking and is a GUI application.

Signed-off-by: Stefan Hajnoczi <stefanha@jammr.net>

### CODE REVIEW

File: net.jammr.jammr.json [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        "--share=network",
+        "--device=all"
+    ],
+    "modules": [
+        {
+            "name": "icons",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -Dm644 jammr.png /app/share/icons/hicolor/48x48/apps/net.jammr.jammr.png",
+                "install -Dm644 jammr.svg /app/share/icons/hicolor/128x128/apps/net.jammr.jammr.svg"
```
Comment (bilelmoussaoui) [inline net.jammr.jammr.json:21]:
```suggestion
                "install -Dm644 jammr.svg /app/share/icons/hicolor/scalable/apps/net.jammr.jammr.svg"
```

File: net.jammr.jammr.json [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                    "tag": "jammr-1.2.8"
+                },
+                {
+                    "type": "patch",
+                    "path": "patches/wahjam/avoid-zero-length-vla.patch"
+                }
+            ]
+        }
+    ],
+    "cleanup": [
```
Comment (bilelmoussaoui) [inline net.jammr.jammr.json:128]: Nitpick: the cleanup by convention goes just after the finish-args

File: net.jammr.jammr.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bilelmoussaoui) [inline net.jammr.jammr.appdata.xml:1]: The appdata, desktop file and icons should be part of the release tarball if possible.

File: net.jammr.jammr.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        <category>Music</category>
+    </categories>
+    <url type="homepage">https://jammr.net/</url>
+    <url type="faq">https://jammr.net/faq.html</url>
+    <url type="help">https://forum.jammr.net/1/</url>
+    <url type="contact">https://jammr.net/contact.html</url>
+    <launchable type="desktop-id">net.jammr.jammr.desktop</launchable>
+    <project_license>GPL-2.0-or-later</project_license>
+    <developer_name>jammr ltd</developer_name>
+    <releases>
```
Comment (bilelmoussaoui) [inline net.jammr.jammr.appdata.xml:27]: You need OARS tag https://hughsie.github.io/oars/

File: net.jammr.jammr.appdata.xml (hunk 3 above)
Comment (stefanha) [reply net.jammr.jammr.appdata.xml:1]: Understood.  This initial manifest is for the existing jammr 1.2.8 release so I can't add these files retrospectively.

I will merge the appdata, desktop file, and icons into wahjam.git in the next release and then they can be removed from the manifest.

Hope it's okay to include them here until the next jammr release.

File: net.jammr.jammr.appdata.xml (hunk 4 above)
Comment (stefanha) [reply net.jammr.jammr.appdata.xml:27]: Thanks, will add in the next revision.

File: net.jammr.jammr.json (hunk 2 above)
Comment (stefanha) [reply net.jammr.jammr.json:128]: Thanks, will move in the next revision.

**Comment by stefanha** [comment]
@bilelmoussaoui Thanks for the review!  I have pushed another revision with your comments addressed.

File: net.jammr.jammr.json [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        {
+            "name": "appdata",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -Dm644 net.jammr.jammr.appdata.xml /app/share/metainfo/net.jammr.jammr.appdata.xml"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "path": "net.jammr.jammr.appdata.xml"
```
Comment (LeandroStanger) [inline net.jammr.jammr.json:50]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

**Comment by bilelmoussaoui** [comment]
bot, build net.jammr.jammr

**Comment by stefanha** [comment]
@LeandroStanger Got it.  I have pushed a new revision that uses metainfo instead of appdata.

**Comment by stefanha** [comment]
I see the build failed because there are no screenshots.  I will add a screenshot and try again.

**Comment by stefanha** [comment]
@LeandroStanger @bilelmoussaoui Screenshot added.

**Comment by nedrichards** [comment]
bot, build net.jammr.jammr

@stefanha (you can trigger builds yourself this way if you wish as well)

**Comment by nedrichards** [comment]
Still needs screenshots, which I am sure you are aware.

**Comment by stefanha** [comment]
Oops, I mistyped the <screenshots> XML tag. Fixed now.

**Comment by stefanha** [comment]
bot, build net.jammr.jammr

**Comment by stefanha** [comment]
I have successfully tested Build 19413.  Looks good!

**Pull request closed without merging**